Error when building the DocumentProcessing sample

hi,

I tried to build your DocumentProcessing sample, and got this error (on Fedora 40, with .NET SDK 9.0.103, latest commit on main branch at the time of writing i.e. 18a2f43):

```
/home/vito/Desktop/test_task/DTasks/src/DTasks.Core/BackgroundDAwaitable.cs(106,71): error CS0305: Using the generic type 'BackgroundDTask<TTask>' requires 1 type arguments
```

After commenting out the error line, everything worked as expected.

I stepped through the code to understand how your serialization/deserialization code of the async/await state machine worked: congrats, very cool work of reverse engineering!

Vito

--- a/src/DTasks.Core/BackgroundDAwaitable.cs
+++ b/src/DTasks.Core/BackgroundDAwaitable.cs
@@ -103,7 +103,7 @@ internal sealed class BackgroundDTask<TTask>(IDAsyncRunnable runnable) : DTask<T
 
     protected override void Run(IDAsyncRunner runner)
     {
-        Debug.Fail($"'{nameof(Run)}' should not be invoked on {nameof(BackgroundDTask<>)}.");
+        Debug.Fail($"'{nameof(Run)}' should not be invoked on {nameof(BackgroundDTask<DTask>)}.");
     }
 
     void IDAsyncResultBuilder<TTask>.SetResult(TTask result)

